src/core/public/http/external_url_service.test.ts

```diff
@@ -0,0 +1,212 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ExternalUrlConfig } from 'src/core/server/types';
+
+import { injectedMetadataServiceMock } from '../mocks';
+
+import { ExternalUrlService } from './external_url_service';
+
+const setupService = ({
+  location,
+  serverBasePath,
+  policy,
+}: {
+  location: URL;
+  serverBasePath: string;
+  policy: ExternalUrlConfig['policy'];
+}) => {
+  const injectedMetadata = injectedMetadataServiceMock.createSetupContract();
+  injectedMetadata.getExternalUrlConfig.mockReturnValue({ policy });
+  injectedMetadata.getServerBasePath.mockReturnValue(serverBasePath);
+
+  const service = new ExternalUrlService();
+  return {
+    setup: service.setup({
+      injectedMetadata,
+      location,
+    }),
+  };
+};
+
+describe('External Url Service', () => {
+  describe('#validateUrl', () => {
+    describe('internal requests with a server base path', () => {
```

Comment:
I did some tinkering using the [protocol resolution bypass](https://owasp.org/www-community/xss-filter-evasion-cheatsheet#protocol-resolution-bypass) technique. Here are some extra test cases you could add.

IMO it is working as intended if it treats `//<hostname>` as a host and attempts to validate accordingly. It's not likely that a bad actor could use protocol resolution bypass when a basepath is present, because of our attack surface, but it could feasibly happen when no basepath is used, depending on how consumers are handling user input.

<details><summary>See diff</summary>

```diff
diff --git a/src/core/public/http/external_url_service.test.ts b/src/core/public/http/external_url_service.test.ts
index 8aae656d22b..59415b7de31 100644
--- a/src/core/public/http/external_url_service.test.ts
+++ b/src/core/public/http/external_url_service.test.ts
@@ -94,6 +94,36 @@ describe('External Url Service', () => {
 
         expect(result).toBeNull();
       });
+
+      describe('handles protocol resolution bypass', () => {
+        it('does not allow relative URLs that include a host', () => {
+          const { setup } = setupService({ location, serverBasePath, policy: [] });
+          const urlCandidate = `/some/path?foo=bar`;
+          const result = setup.validateUrl(`//www.google.com${serverBasePath}${urlCandidate}`);
+
+          expect(result).toBeNull();
+        });
+
+        it('does allow relative URLs that include a host if allowed by policy', () => {
+          const { setup } = setupService({
+            location,
+            serverBasePath,
+            policy: [
+              {
+                allow: true,
+                host: 'www.google.com',
+              },
+            ],
+          });
+          const urlCandidate = `/some/path?foo=bar`;
+          const result = setup.validateUrl(`//www.google.com${serverBasePath}${urlCandidate}`);
+
+          expect(result).toBeInstanceOf(URL);
+          expect(result?.toString()).toEqual(
+            `https://www.google.com${serverBasePath}${urlCandidate}`
+          );
+        });
+      });
     });
 
     describe('internal requests without a server base path', () => {
@@ -119,6 +149,34 @@ describe('External Url Service', () => {
         expect(result).toBeInstanceOf(URL);
         expect(result?.toString()).toEqual(`${kibanaRoot}/some/path?foo=bar`);
       });
+
+      describe('handles protocol resolution bypass', () => {
+        it('does not allow relative URLs that include a host', () => {
+          const { setup } = setupService({ location, serverBasePath, policy: [] });
+          const urlCandidate = `/some/path?foo=bar`;
+          const result = setup.validateUrl(`//www.google.com${urlCandidate}`);
+
+          expect(result).toBeNull();
+        });
+
+        it('allows relative URLs that include a host in the allow list', () => {
+          const { setup } = setupService({
+            location,
+            serverBasePath,
+            policy: [
+              {
+                allow: true,
+                host: 'www.google.com',
+              },
+            ],
+          });
+          const urlCandidate = `/some/path?foo=bar`;
+          const result = setup.validateUrl(`//www.google.com${urlCandidate}`);
+
+          expect(result).toBeInstanceOf(URL);
+          expect(result?.toString()).toEqual(`https://www.google.com${urlCandidate}`);
+        });
+      });
     });
 
     describe('external requests', () => {
```

</details>